examples/interaction.rs

```diff
@@ -0,0 +1,323 @@
+                        .or()
+                        .or(),
+                    )
+                    .property(Expectation::Eventually, "success", |_, state| {
+                        state.actor_states.iter().any(|s| state_filter_success(s))
+                    })
+                    .checker()
+                    .threads(num_cpus::get())
+                    .target_max_depth(30) // To ensure termination of the checker, since the state space is very loosely bounded.
+                    .spawn_bfs()
+                    .report(&mut WriteReporter::new(&mut std::io::stdout()));
+
+            checker.assert_properties();
+        }
+
+        Some("explore") => {
+            let checker =
+                ActorModel::<choice![ExternalInput, Supervisor, Counter], (), u8>::new((), 0)
+                    .actor(Choice::new(ExternalInput {
+                        threshold: 3,
+                        supervisor_addr: 1.into(),
+                    }))
+                    .actor(
+                        Choice::new(Supervisor {
+                            initial_state: SupervisorState {
+                                addr: 1.into(),
+                                threshold: 3,
+                                counter_addr: 2.into(),
+                                input_addr: 0.into(),
+                                success: false,
+                            },
+                        })
+                        .or(),
+                    )
+                    .actor(
+                        Choice::new(Counter {
+                            initial_state: CounterState {
+                                addr: 1.into(),
+                                counter: 0,
+                            },
+                        })
+                        .or()
+                        .or(),
+                    )
+                    .property(Expectation::Eventually, "success", |_, state| {
+                        state.actor_states.iter().any(|s| state_filter_success(s))
+                    })
+                    .checker()
+                    .threads(num_cpus::get())
+                    .target_max_depth(30); // To ensure termination of the checker, since the state space is very loosely bounded.
+
+            checker.serve("0:3000");
+        }
+
+        _ => {
+            println!("USAGE:");
+            println!("  ./interaction check");
+            println!("  ./interaction explore");
+        }
+    }
+    Ok(())
+}
+
+// We use this helper function to improve readability of check_discovery.
+// It also shows how to use the choice! macro for destructuring.
+fn state_filter_success(s: &Arc<choice![InputState, SupervisorState, CounterState]>) -> bool {
+    match s.as_ref() {
+        choice!(0 -> _v) => false, // InputState
+        choice!(1 -> v) => {
+            if v.success == true {
+                true
+            } else {
+                false
+            }
+        } // SupervisorState
```

Comment:
```suggestion
        choice!(1 -> v) => v.success, // SupervisorState
```

Reply:
Thanks! That was a debug artifact I forgot